source/auth/auth.rst

```diff
@@ -409,6 +473,58 @@ mechanism_properties
 	MUST NOT be specified.
 
 
+SCRAM-SHA-256
+~~~~~~~~~~~~~
+
+:since: 4.0
+
+SCRAM-SHA-256 extends `RFC 5802 <http://tools.ietf.org/html/rfc5802>`_ and
+is formally defined in `RFC 7677 <https://tools.ietf.org/html/rfc7677>`_.
+
+The MongoDB SCRAM-SHA-256 mechanism works similarly to the SCRAM-SHA-1
+mechanism, with the following changes:
+
+- The SCRAM algorithm MUST use SHA-256 as the hash function instead of SHA-1.
+- User names MUST be prepared with SASLprep, per RFC 5802.
+- Passwords MUST be prepared with SASLprep, per RFC 5802.  Passwords are
+  used directly for key derivation ; they MUST NOT be digested as they are in
+  SCRAM-SHA-1.
+
+Additionally, drivers MUST enforce a minimum iteration count of 4096 and MUST
+error if the authentication conversation specifies a lower count.  This
+mitigates downgrade attacks by a man-in-the-middle attacker.
```

Comment:
In the Perl driver CR, @behackett pointed out that this paragraph needs to apply to SCRAM-SHA-1 as well.

Reply:
Done.